eng/common/TestResources/Get-TestSecrets.ps1

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env pwsh
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+#Requires -Version 6.0
+#Requires -PSEdition Core
+#Requires -Modules @{ModuleName='Az.Accounts'; ModuleVersion='1.6.4'}
+#Requires -Modules @{ModuleName='Az.KeyVault'; ModuleVersion='4.6.0'}
+
+[CmdletBinding()]
+[OutputType([pscustomobject])]
+param (
+    [Parameter(Mandatory=$true, Position=0)]
+    [string] $VaultName,
+
+    [Parameter(Mandatory=$true, Position=1)]
+    [string] $Name
+)
+
+# By default stop for any error.
+if (!$PSBoundParameters.ContainsKey('ErrorAction')) {
+    $ErrorActionPreference = 'Stop'
+}
+
+$secret = Get-AzKeyVaultSecret -VaultName:$VaultName -Name:$Name -ErrorAction SilentlyContinue -ErrorVariable err
+if ($err) {
+    if ($err.Exception.SocketErrorCode -eq 'HostNotFound') {
+        Write-Error "Key Vault $VaultName not found. Use 'Set-AzContext -SubscriptionId {subscriptionId}' to select appropriate subscription."
+        return
```

Comment:
If this returns like this while piped to `New-TestResources.ps1` I assume the script will continue running without any keyvault values? Or will it fail due to no object passed through?

Reply:
With errors, `SilentlyContinue` won't affect the cmdlet returning anything or not (at least, if well-behaved) so `$secret` would still be null/empty (same thing in PowerShell, since it recognizes "truthiness" like JS). `SilentlyContinue` just lets us do error handling, and being explicit here means the `-ea` passed to this script itself will be ignored. It can apply elsewhere for all I care, but we definitely want to fail in places where I did early returns (or succeed, as further below).